src/main/java/com/rybka/view/ExchangeView.java

```diff
@@ -43,8 +47,13 @@ public void showDialog() {
 
             showExchange(convertedResult);
 
-        } catch (Exception e) {
-            log.error("Some issues are occurred! Reason: " + e.getMessage());
+            exportService.export(currencyHistoryDAO.findAll());
+        } catch (NullPointerException exception) {
+            log.error("Invalid currency for exchange.");
```

Comment:
I believe we should rethink hole catch logic ...

Reply:
ok